src/main/kotlin/lotto/domain/LottoPrizePolicy.kt

```diff
@@ -0,0 +1,16 @@
+package lotto.domain
+
+class LottoPrizePolicy(
+    val wonMatchedCount: Int,
+    val wonPrize: Money
+) {
+
+    fun isWon(lottoTicket: LottoTicket, winningLottoNumbers: LottoTicketNumbers): Boolean {
+        return won(lottoTicket, winningLottoNumbers) != null
+    }
+
+    fun won(lottoTicket: LottoTicket, winningLottoNumbers: LottoTicketNumbers): Money? {
+        if (lottoTicket.getMatchedCount(winningLottoNumbers) == wonMatchedCount) return wonPrize
+        return null
+    }
```

Comment:
`isWon` 에서 `won`을 호출하는 것은 이겼는지 확인하기 위해 이겼을때 호출하는 기능을 사용한다는 것으로 보여 어색해보입니다.

아래와 같이 확장함수를 이용해서 표현해보는건 어떨까요? 

```suggestion
    fun isWon(lottoTicket: LottoTicket, winningLottoNumbers: LottoTicketNumbers): Boolean {
        return lottoTicket.isMatch(winningLottoNumbers)
    }

    fun won(lottoTicket: LottoTicket, winningLottoNumbers: LottoTicketNumbers): Money {
        return wonPrize
    }

    private fun LottoTicket.isMatch(winningLottoNumbers: LottoTicketNumbers) = this.getMatchedCount(winningLottoNumbers) == wonMatchedCount
```

또한 이미 `isWon`에서 해당 `won`을 호출할수 있는 조건을 검증했는데 한번더 검사를 해야하는지 의문입니다. 이 객체에게 메세지를 던지는 쪽에서 `isWon` 후 `won`을 호출하도록 정의해주는 것이 자연스러울것 같습니다.